test/testenv/appframework_utils.go

```diff
@@ -311,3 +325,300 @@ func WaitforPhaseChange(deployment *Deployment, testenvInstance *TestEnv, name s
+				VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, phase, appFileList2ndScopeCluster)
+			} else {
+				VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, phase, appFileList)
+			}
+		}
+		if appSource.CrKind == "Standalone" || appSource.CrKind == "MonitoringConsole" {
+			testenvInstance.Log.Info("Verify apps 'download' state on %v CR", appSource.CrKind)
+			VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, phase, appFileList)
+		}
+	}
+
+	// Verify apps 'PodCopy' state for all CRs
+	phase = enterpriseApi.PhasePodCopy
+	for _, appSource := range appSource {
+		if appSource.CrKind == "ClusterMaster" || appSource.CrKind == "SearchHeadCluster" {
+			testenvInstance.Log.Info("Verify apps 'PodCopy' state on %v CR", appSource.CrKind)
+			if appFileList2ndScopeCluster != nil && appList2ndScopeCluster != nil && appSource.CrAppScope == "cluster" {
+				VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, phase, appFileList2ndScopeCluster)
+			} else {
+				VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, phase, appFileList)
+			}
+		}
+		if appSource.CrKind == "Standalone" || appSource.CrKind == "MonitoringConsole" {
+			testenvInstance.Log.Info("Verify apps 'PodCopy' state on %v CR", appSource.CrKind)
+			VerifyAppListPhase(deployment, testenvInstance, appSource.CrName, appSource.CrKind, appSource.CrAppSourceName, phase, appFileList)
+		}
+	}
+
+	// Verify apps packages are deleted from the operator pod for all CRs
+	opPod := GetOperatorPodName(testenvInstance.GetName())
+	for _, appSource := range appSource {
+		if appSource.CrKind == "ClusterMaster" || appSource.CrKind == "SearchHeadCluster" {
+			testenvInstance.Log.Info(fmt.Sprintf("Verify apps %s packages are deleted from the operator pod for %v", appVersion, appSource.CrKind))
+			if appFileList2ndScopeCluster != nil && appList2ndScopeCluster != nil && appSource.CrAppScope == "cluster" {
+				opPath := filepath.Join(splcommon.AppDownloadVolume, "downloadedApps", testenvInstance.GetName(), appSource.CrKind, deployment.GetName(), enterpriseApi.ScopeCluster, appSource.CrAppSourceName)
+				VerifyAppsPackageDeletedOnContainer(deployment, testenvInstance, testenvInstance.GetName(), []string{opPod}, appFileList2ndScopeCluster, opPath)
+			} else if appSource.CrAppScope == "cluster" {
+				opPath := filepath.Join(splcommon.AppDownloadVolume, "downloadedApps", testenvInstance.GetName(), appSource.CrKind, deployment.GetName(), enterpriseApi.ScopeCluster, appSource.CrAppSourceName)
+				VerifyAppsPackageDeletedOnContainer(deployment, testenvInstance, testenvInstance.GetName(), []string{opPod}, appFileList, opPath)
+			} else if appSource.CrAppScope == "local" {
```

Comment:
Conditions for `appSource.CrAppScope == "cluster"` and `appSource.CrAppScope == "cluster"` are used to differentiate path. However this distinction is already achievable by using `appSource.CrAppScope`